src/engine/renderRouter.ts

```diff
@@ -16,16 +16,60 @@ const EMPTY_RENDER_FUNCTIONS: RenderFunctions = {
   renderWled: async () => create(WledRenderTargetSchema, {}),
 };
 
+const FPS_BUFFER_SIZE = 100;
+
 let renderFunctions = EMPTY_RENDER_FUNCTIONS;
 const dmxSubscriptions: Map<
   bigint,
-  Array<(o: DmxRenderOutput) => void>
+  Array<(o: DmxRenderOutput, fps: number) => void>
 > = new Map();
 const wledSubscriptions: Map<
   bigint,
-  Array<(o: WledRenderTarget) => void>
+  Array<(o: WledRenderTarget, fps: number) => void>
 > = new Map();
 
+// FPS tracking state
+const dmxLastRenderTime: Map<bigint, number> = new Map();
+const dmxFpsBuffers: Map<bigint, number[]> = new Map();
+const wledLastRenderTime: Map<bigint, number> = new Map();
+const wledFpsBuffers: Map<bigint, number[]> = new Map();
```

Comment:
Instead of having multiple maps, lets only have one map of `bigint` to an array of render times. Upon each render we can re-calculate the difference between all render times to determine an FPS. Furthermore, for FPS tracking the WLED and DMX FPS buffers are exactly the same and don't need to be distinguished per map.